docs/features.md

```diff
@@ -874,30 +874,100 @@ For the `AuthConfig` above,
+_Conditions_, named `when` in the AuthConfig API, are logical expressions, composed of patterns and logical operator AND and OR, that can be used to condition the evaluation of a particular auth rule within an AuthConfig, as well as of the AuthConfig altogether ("top-level conditions").
 
-The scope for a set of `when` conditions can be the entire `AuthConfig` ("top-level conditions") or a particular evaluator of any phase of the auth pipeline.
+The patterns are evaluated against the [Authorization JSON](./architecture.md#the-authorization-json), where each pattern is a tuple composed of:
+- `selector`: a [JSON path](#common-feature-json-paths-selector) to fetch a value from the Authorization JSON
+- `operator`: one of: `eq` (_equals_); `neq` (_not equal_); `incl` (_includes_) and `excl` (_excludes_), for when the value fetched from the Authorization JSON is expected to be an array; `matches`, for regular expressions
+- `value`: a static string value to compare the value selected from the Authorization JSON with.
 
-Each expression is a tuple composed of:
-- a `selector`, to fetch from the Authorization JSON – see [Common feature: JSON paths](#common-feature-json-paths-selector) for details about syntax;
-- an `operator` – `eq` (_equals_), `neq` (_not equal_); `incl` (_includes_) and `excl` (_excludes_), for arrays; and `matches`, for regular expressions;
-- a fixed comparable `value`
+An expression contains one or more patterns and they must either all evaluate to true ("AND" operator, declared by grouping the patterns within an `all` block) or at least one of the patterns must be true ("OR" operator, when grouped within an `any` block.) Patterns not explicitly grouped are AND'ed by default.
+
+To avoid repetitions when listing patterns, any set of literal `{ pattern, operator, value }` tuples can be stored at the top-level of the AuthConfig spec, indexed by name, and later referred within an expression by including a `patternRef` in the block of conditions.
+
+**Examples of `when` conditions**
+
+i) to skip an entire `AuthConfig` based on the context (AND operator assumed by default):
+
+```yaml
+spec:
+  when: # auth enforced only on requests to POST /resources/*
+  - selector: context.request.http.method
+    operator: eq
+    value: POST
+  - selector: context.request.http.path
+    operator: matches
+    value: ^/resources/.*
+```
+
+ii) equivalent to the above with explicit AND operator (i.e., `all` block):
+
+```yaml
+spec:
+  when: # auth enforced only on requests to POST /resources/*
+  - all:
+    - selector: context.request.http.method
+      operator: eq
+      value: POST
+    - selector: context.request.http.path
+      operator: matches
+      value: ^/resources/.*
+```
+
+iii) OR condition (i.e., `any` block):
+
+```yaml
+spec:
+  when: # auth enforced only on requests with HTTP method equals to POST or PUT
+  - any:
+    - selector: context.request.http.method
```

Comment:
side point: Have you considered and `in` operator as this would (at least from an API perspective) improve the verbosity. 
```
 - selector: context.request.http.method
    operator: in
    values: ["PUT","POST"]
```

Reply:
I guess you could also do this with `regexp`?

Reply:
https://github.com/Kuadrant/authorino/issues/357